x-pack/plugins/security_solution/public/users/containers/users/details/index.tsx

```diff
@@ -53,104 +41,53 @@ export const useUserDetails = ({
-    refetch: refetch.current,
-    startDate,
+    errorMessage: i18n.FAIL_USER_DETAILS,
+    abort: skip,
   });
 
-  const userDetailsSearch = useCallback(
-    (request: UserDetailsRequestOptions | null) => {
-      if (request == null || skip) {
-        return;
-      }
-
-      const asyncSearch = async () => {
-        abortCtrl.current = new AbortController();
-        setLoading(true);
-
-        searchSubscription$.current = data.search
-          .search<UserDetailsRequestOptions, UserDetailsStrategyResponse>(request, {
-            strategy: 'securitySolutionSearchStrategy',
-            abortSignal: abortCtrl.current.signal,
-          })
-          .subscribe({
-            next: (response) => {
-              if (isCompleteResponse(response)) {
-                setLoading(false);
-                setUserDetailsResponse((prevResponse) => ({
-                  ...prevResponse,
-                  userDetails: response.userDetails,
-                  inspect: getInspectResponse(response, prevResponse.inspect),
-                  refetch: refetch.current,
-                }));
-                searchSubscription$.current.unsubscribe();
-              } else if (isErrorResponse(response)) {
-                setLoading(false);
-                addWarning(i18n.ERROR_USER_DETAILS);
-                searchSubscription$.current.unsubscribe();
-              }
-            },
-            error: (msg) => {
-              setLoading(false);
-              addError(msg, {
-                title: i18n.FAIL_USER_DETAILS,
-              });
-              searchSubscription$.current.unsubscribe();
-            },
-          });
-      };
-      searchSubscription$.current.unsubscribe();
-      abortCtrl.current.abort();
-      asyncSearch();
-      refetch.current = asyncSearch;
-    },
-    [data.search, addError, addWarning, skip]
+  const userDetailsResponse = useMemo(
+    () => ({
+      endDate,
+      userDetails: response.userDetails,
+      id,
+      inspect,
+      isInspected: false,
+      refetch,
+      startDate,
+    }),
+    [endDate, id, inspect, refetch, response.userDetails, startDate]
   );
 
-  useEffect(() => {
-    setUserDetailsRequest((prevRequest) => {
-      const myRequest = {
-        ...(prevRequest ?? {}),
-        defaultIndex: indexNames,
-        factoryQueryType: UsersQueries.details,
-        userName,
-        timerange: {
-          interval: '12h',
-          from: startDate,
-          to: endDate,
-        },
-      };
-      if (!deepEqual(prevRequest, myRequest)) {
-        return myRequest;
-      }
-      return prevRequest;
-    });
-  }, [endDate, userName, indexNames, startDate]);
+  const userDetailsRequest = useMemo(
+    () => ({
+      defaultIndex: indexNames,
+      factoryQueryType: UsersQueries.details,
+      userName,
+      timerange: {
+        interval: '12h',
+        from: startDate,
+        to: endDate,
+      },
+    }),
+    [endDate, indexNames, startDate, userName]
+  );
```

Comment:
Is there a reason why we don't do the `deepEqual` check to the request no?

Reply:
I think these values are all string, so we could relying on useMemo's reference and its dependency array to avoid re-render.